File: WebViewDialog/app/src/main/java/com/eypcnn/webviewdialog/MainActivity.java
package com.eypcnn.webviewdialog;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import android.webkit.WebView;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.webViewDialog);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final Dialog dialog = new Dialog(MainActivity.this);
                dialog.setContentView(R.layout.custom_dialog);

                WebView webView = dialog.findViewById(R.id.webView);
                String webviewdata = "<html><body style='background:black;color:white;padding:2em;'> Custom WebView Dialog </body></html>";
                webView.loadData(webviewdata.toString(), "text/html", "utf-8");

                Button close = dialog.findViewById(R.id.diloagClose);
                close.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        dialog.dismiss();
                    }
                });

                dialog.show();
            }
        });
    }
}

File: WebViewDialogKotlin/app/src/main/java/com/eypcnn/webviewdialog/MainActivity.kt
package com.eypcnn.webviewdialog

import android.app.Dialog
import android.os.Bundle
import android.webkit.WebView
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button = findViewById<Button>(R.id.webViewDialog)
        button.setOnClickListener {
            val dialog = Dialog(this@MainActivity)
            dialog.setContentView(R.layout.custom_dialog)

            val webView = dialog.findViewById<WebView>(R.id.webView)
            val webviewdata = "<html><body style='background:black;color:white;padding:3em;'> Custom WebView Dialog Kotlin </body></html>"
            webView.loadData(webviewdata, "text/html", "utf-8")

            val close = dialog.findViewById<Button>(R.id.diloagClose)
            close.setOnClickListener { dialog.dismiss() }

            dialog.show()
        }

    }
}